src/commutator_rpnl.F

```diff
@@ -398,4 +402,509 @@ SUBROUTINE build_com_rpnl(matrix_rv, qs_kind_set, sab_orb, sap_ppnl, eps_ppnl)
+!> \param matrix_rxrv ...
+!> \param matrix_rrv ...
+!> \param ref_point ...
+!> \param particle_set ...
+!> \param cell ...
+! **************************************************************************************************
+   SUBROUTINE build_com_mom_nl(qs_kind_set, sab_all, sap_ppnl, eps_ppnl, matrix_rv, matrix_rxrv, matrix_rrv, ref_point, &
+                               particle_set, cell)
+
+      TYPE(qs_kind_type), DIMENSION(:), POINTER          :: qs_kind_set
+      TYPE(neighbor_list_set_p_type), DIMENSION(:), &
+         POINTER                                         :: sab_all, sap_ppnl
+      REAL(KIND=dp), INTENT(IN)                          :: eps_ppnl
+      TYPE(dbcsr_p_type), DIMENSION(:), OPTIONAL, &
+         POINTER                                         :: matrix_rv, matrix_rxrv, matrix_rrv
+      REAL(KIND=dp), DIMENSION(3), INTENT(IN), OPTIONAL  :: ref_point
+      TYPE(particle_type), DIMENSION(:), OPTIONAL, &
+         POINTER                                         :: particle_set
+      TYPE(cell_type), OPTIONAL, POINTER                 :: cell
+
+      CHARACTER(LEN=*), PARAMETER :: routineN = 'build_com_mom_nl', &
+         routineP = moduleN//':'//routineN
+
+      INTEGER :: handle, i, iab, iac, iatom, ibc, icol, ikind, ilist, ind, inode, irow, iset, &
+         jatom, jkind, jneighbor, kac, katom, kbc, kkind, l, lc_max, lc_min, ldai, ldsab, lppnl, &
+         maxco, maxder, maxl, maxlgto, maxlppnl, maxppnl, maxsgf, mepos, na, nb, ncoa, ncoc, &
+         nkind, nlist, nneighbor, nnode, np, nppnl, nprjc, nseta, nsgfa, nthread, order, prjc, sgfa
+      INTEGER, DIMENSION(3)                              :: cell_b, cell_c
+      INTEGER, DIMENSION(:), POINTER                     :: la_max, la_min, npgfa, nprj_ppnl, &
+                                                            nsgf_seta
+      INTEGER, DIMENSION(:, :), POINTER                  :: first_sgfa
+      LOGICAL                                            :: asso_rrv, asso_rv, asso_rxrv, found, go, &
+                                                            gpot, my_ref, my_rrv, my_rv, my_rxrv, &
+                                                            ppnl_present, spot
+      REAL(KIND=dp)                                      :: dac, ppnl_radius
+      REAL(KIND=dp), ALLOCATABLE, DIMENSION(:, :, :)     :: ai_work, sab, work
+      REAL(KIND=dp), DIMENSION(1)                        :: rprjc, zetc
+      REAL(KIND=dp), DIMENSION(3)                        :: ra, rab, rac, raf, rc, rcf, rf
+      REAL(KIND=dp), DIMENSION(:), POINTER               :: alpha_ppnl, set_radius_a
+      REAL(KIND=dp), DIMENSION(:, :), POINTER            :: cprj, rpgfa, sphi_a, vprj_ppnl, zeta
+      REAL(KIND=dp), DIMENSION(:, :, :), POINTER         :: achint, acint, bchint, bcint
+      TYPE(alist_type), POINTER                          :: alist_ac, alist_bc
+      TYPE(block_p_type), DIMENSION(:), POINTER          :: blocks_rrv, blocks_rv, blocks_rxrv
```

Comment:
I think it should be possible to use `ALLOCATABLE` here instead of `POINTER`, might be true for others as well